aten/src/ATen/core/dispatch/Dispatcher.h

```diff
@@ -346,25 +343,12 @@ class TORCH_API OperatorHandle {
   }
 
 private:
-  explicit OperatorHandle(std::list<Dispatcher::OperatorDef>::iterator operatorIterator)
-  : operatorDef_(&*operatorIterator), operatorIterator_(operatorIterator)  {}
+  explicit OperatorHandle(const impl::OperatorEntry *entry) : operatorEntry_(entry) {}
+  explicit OperatorHandle(const Dispatcher::OperatorCleanup &op) : operatorEntry_(&op->op) {}
   friend class Dispatcher;
   template<class> friend class TypedOperatorHandle;
 
-  // Storing a direct pointer to the OperatorDef even though we
-  // already have the iterator saves an instruction in the critical
-  // dispatch path. The iterator is effectively a
-  // pointer-to-std::list-node, and (at least in libstdc++'s
-  // implementation) the element is at an offset 16 bytes from that,
-  // because the prev/next pointers come first in the list node
-  // struct. So, an add instruction would be necessary to convert from the
-  // iterator to an OperatorDef*.
-  Dispatcher::OperatorDef* operatorDef_;
-
-  // We need to store this iterator in order to make
-  // Dispatcher::cleanup() fast -- it runs a lot on program
-  // termination (and presuambly library unloading).
-  std::list<Dispatcher::OperatorDef>::iterator operatorIterator_;
+  const impl::OperatorEntry* operatorEntry_ = nullptr;
```

Comment:
Does this actually safe anything? After compilation, `std::list<Dispatcher::OperatorDef>::iterator` is just a `Dispatcher::OperatorDef*` which in turn is just a `impl::OperatorEntry*` because OperatorEntry is stored by value as a first struct member of OperatorDef.

So the only effect I'm seeing from those changes is to remove the `operatorDef_` member, which is actually not even needed in the old implementation, but was specifically added to improve perf, see the comment there.

Reply:
You are very right. My bad, I missed that `impl::OperatorEntry` is stored by value.
It saves 3x pointers per callback, but I guess most ppl don't care about a few KBs anymore.